src/utils/actions.js

```diff
@@ -230,6 +230,77 @@ const actions = [
     },
     requiredInput: ['partial'],
   },
+  {
+    /** *************************************
+     * Execute Automation contacts internal urls to obtain automation
+     * information and execute the automation selected in the property panel in
+     * edit mode.
+     *
+     * ARGS
+     * app - Reference to current app inherited from index.js
+     * automation - the item id of the automation to contact the items api
+     * automationPostData - boolean value. If true, creates a bookmark of the
+     * current selections and sends the resulting bookmark id as an input
+     * parameter to the selected automation.
+    */
+
+    label: 'Execute Automation',
+    value: 'executeAutomation',
+    getActionCall: ({ app, automation, automationPostData }) => async () => {
+      if (automation !== undefined) {
+        try {
+          const itemInfo = await fetch(`../api/v1/items/${automation}`)
+            .then(response => response.json());
+          const autoInfo = await fetch(`../api/v1/automations/${itemInfo.resourceId}`)
+            .then(response => response.json());
+          let executePath = `../api/v1/automations/${autoInfo.guid}/actions/execute?X-Execution-Token=${autoInfo.execution_token}`;
+          if (automationPostData && autoInfo.hasinputs) {
+            const inputBlock = await fetch(`../api/v1/automations/${itemInfo.resourceId}/blocks`)
+              .then(response => response.json())
+              .then(blocks => {
+                let item;
+                // Used a for loop because I read on the Internet it's faster than
+                // using array operations
+                for (let i = 0; i < blocks.blocks.length; i++) {
+                  if (blocks.blocks[i].displayName === 'Inputs') {
+                    item = blocks.blocks[i];
+                    break;
+                  }
+                }
+                return item.form[0].label.toLowerCase();
+              });
+            const newDate = new Date();
+            const bmkProp = {
```

Comment:
Would these bookmarks show up in the bookmark overview as well? Do they need to be cleaned up in any way?

Reply:
Is there a way to create hidden bookmarks still? I'd prefer to do that but if it is possible it's not documented :) Otherwise cleanup would have to happen by the automation if the automation owner adds blocks to remove the bookmark.